### ISSUE TITLE
Kapt Compile Avoidance and AndroidX

### ISSUE THREAD

**Issue by smuldr** [bumptech/glide#3606]
**Glide Version**: 4.9.0

**Other tools:**
- Kotlin 1.3.30
- AndroidX 1.0.0 (with Jetifier)

**Issue details / Repro steps / Use case background**: 

Glide fails when enabling Gradle's compile avoidance for Kapt as explained in the [releases notes for Kotin 1.3.30](https://blog.jetbrains.com/kotlin/2019/04/kotlin-1-3-30-released/):

```properties
kapt.include.compile.classpath=false
```

It looks like _some_ (?) sources from Glide's annotation processor are not passed through the Jetifier. 

For example: the `GlideApp` class has a mix of AndroidX and legacy Android Support annotations in its imports:
```java
package com.example.glide;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;
import android.view.View;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.VisibleForTesting;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import java.io.File;
import java.lang.Deprecated;
import java.lang.String;

/**
 * The entry point for interacting with Glide for Applications
 *
 * <p>Includes all generated APIs from all
 * {@link com.bumptech.glide.annotation.GlideExtension}s in source and dependent libraries.
 *
 * <p>This class is generated and should not be modified
 * @see Glide
 */
public final class GlideApp {
   ...
}
```

**Stack trace / LogCat**:

Unfortunately, mixing Android Support and AndroidX does not produce working code:
```
com/example/glide/GlideOptions.java:51: error: cannot find symbol
  @NonNull
   ^
  symbol:   class NonNull
  location: class GlideOptions/Users/smuldr/projects/example/app/build/generated/source/kapt/debug/com/example/glide/GlideOptions.java:59: error: cannot find symbol
```

**Comment by mgrzechocinski** [comment]
Same here on Glide 4.8.0. Disabled this new property for the time being.

**Comment by technoir42** [comment]
You can work around this by adding the main Glide artifact to kapt classpath:
```
kapt "com.github.bumptech.glide:compiler:4.9.0"
kapt "com.github.bumptech.glide:glide:4.9.0"
```

**Comment by sjudd** [comment]
I'm not seeing a support annotation in the GlideApp class, nor do I see it referenced in your error.

I do see it in GlideOptions.java:

```
import android.content.res.Resources;
import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.support.annotation.CheckResult;
import android.support.annotation.DrawableRes;
import android.support.annotation.FloatRange;
import android.support.annotation.IntRange;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import com.bumptech.glide.Priority;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.load.Key;
import com.bumptech.glide.load.Option;
import com.bumptech.glide.load.Transformation;
import com.bumptech.glide.load.engine.DiskCacheStrategy;
import com.bumptech.glide.load.resource.bitmap.DownsampleStrategy;
import com.bumptech.glide.request.BaseRequestOptions;
import com.bumptech.glide.request.RequestOptions;
import java.lang.Class;
import java.lang.Cloneable;
import java.lang.Deprecated;
import java.lang.Override;
import java.lang.SafeVarargs;
import java.lang.SuppressWarnings;

/**
 * Automatically generated from {@link com.bumptech.glide.annotation.GlideExtension} annotated classes.
 *
 * @see RequestOptions
 */
@SuppressWarnings("deprecation")
public final class GlideOptions extends RequestOptions implements Cloneable {
```

**Comment by sjudd** [comment]
This isn't simple to fix in Glide, short of removing the annotations from the generated code. Glide's annotation processor already prefers to use androidx annotations for annotations it adds, however it still copies over annotations from the classes the generated classes extend. Short of migrating Glide to androidx, I can't change the annotations on the original classes. 

We could try to map between them, but this is something I'd hope jetifier would do already. 

I think this may be blocked on the general migrate to androidx issue.

**Comment by TWiStErRob** [comment]
@sjudd it may be possible to use the AndroidX annotations everywhere, but keep every other library Support; as a halfway migration. This would resolve all apt related issues with AndroidX I think.

**Comment by sjudd** [comment]
Unfortunately I have the same issue with the annotations that I do with the rest of the androidx libraries. I can't reliably use them in the Google version of Glide. 

The good news is that the migration at Google is in progress. I've also made some progress on migrating to a new way of syncing code that will make it easier to support going to/from androidx as well.

**Comment by sjudd** [comment]
I believe this is fixed by #3184, please re-open if you think otherwise.
